### ISSUE TITLE
False positive with monorepo ESLint plugin package

### ISSUE THREAD

**Issue by Zamiell** [webpro/knip#343]
I think I have found a bug with an unresolved import on knip version 2.40.1.

Steps to reproduce:

```
git clone git@github.com:IsaacScript/isaacscript.git
cd isaacscript
git checkout aa5b9e68ca3ec147d0b87ec83da8bd8a536dcf61
yarn
npx knip
```

Part of the output is as follows:

```
Unresolved imports (11)
eslint-plugin-isaacscript/recommended  .eslintrc.cjs
eslint-plugin-isaacscript/recommended  packages/docs/.eslintrc.cjs
eslint-plugin-isaacscript/recommended  packages/eslint-config-isaacscript/.eslintrc.cjs
eslint-plugin-isaacscript/recommended  packages/eslint-plugin-isaacscript/.eslintrc.cjs
eslint-plugin-isaacscript/recommended  packages/isaac-lua-polyfill/.eslintrc.cjs
eslint-plugin-isaacscript/recommended  packages/isaac-typescript-definitions/.eslintrc.cjs
eslint-plugin-isaacscript/recommended  packages/isaacscript-cli/.eslintrc.cjs
eslint-plugin-isaacscript/recommended  packages/isaacscript-common-node/.eslintrc.cjs
eslint-plugin-isaacscript/recommended  packages/isaacscript-common-ts/.eslintrc.cjs
eslint-plugin-isaacscript/recommended  packages/isaacscript-common/.eslintrc.cjs
eslint-plugin-isaacscript/recommended  packages/isaacscript-spell/.eslintrc.cjs
```

Here, `eslint-plugin-isaacscript` is one of the packages in the monorepo, containing monorepo-specific ESLint rules. This kind of thing is sort of common in JavaScript/TypeScript monorepos.

Thus, this is a false positive, and I believe Knip should not be showing me the output. However, this is kind of a tricky situation, because normally, ESLint plugins are installed in the "package.json" file as a dependency. IMO it doesn't make sense in a monorepo for the root monorepo itself to depend on one of the child packages that it contains, as that would be a dependency cycle. So for my monorepo, the build output is copied to the root "node_modules" folder in a postinstall script. I'm not 100% sure if this is idiomatic in the TypeScript ecosystem, but this was the solution that I landed on after consulting both the official TypeScript and ESLint Discord servers.

So in this case, it's reasonable for Knip to not know that a monorepo subpackage is being manually copied to the root node_modules, but on the other hand, Knip should know that `eslint-plugin-isaacscript` is a subpackage, because I already have a specific workspace entry for it in my knip.json file. So I think Knip should be smart enough to be able to figure this one out on its own. Or barring that, I should be able to specify something in the knip.json file like `internalPackages: ["eslint-plugin-isaacscript"]` to suppress this error.

**Comment by webpro** [comment]
This is a very specific convention of ESLint only: the `/recommended` part is not a path, but a config identifier. Since the `eslint-plugin-isaacscript/recommended` does not lead to a JS file (e.g. `packages/eslint-plugin-isaacscript/recommended.js` does not exist) this is listed as unresolved. What the Knip's eslint plugin should do here is make an exception, but only for the `plugin:isaacscript/recommended` specifier in `extends` options that are prefixed with `plugin:` (because without the `plugin:` it should resolve to a JS file).

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.41.1. See [Release 2.41.1](https://github.com/webpro/knip/releases/tag/2.41.1) for release notes._
